Adding a page with more info about Flutter's typography support (#8215)
flutter/website: src/development/ui/advanced/typography.md

---
title: Flutter's fonts and typography
description: Learn about Flutter's support for typography.
---

> If you aren't familiar with [_typography_][], it essentially

Reviewer: It looks like link check is failing on this link?

> If you aren't familiar with [_typography_][], it essentially
> means how text is styled. [Variable fonts][] (also
> called OpenType fonts), allow you to control various
> aspects of text styling, such as font weight, width, slant,
> and so on.

Reviewer 2: **issues:** The first sentence reads as uncertain. If the reader comes to this page, they either don't know about typography or don't know how it could be applied in Flutter. The description of typography needed some updating as did how variable type works.

Consider:
```suggestion
[_Typography_][] covers the style and appearance of type or fonts.
[Variable fonts][] allow one font to behave like multiple fonts.
You can control most all aspects of type for these fonts including
its weight, width, and variant. The OpenType file format would be
an example of a variable font.
```

## API docs

> Here are some Flutter classes that allow you to
> modify variable fonts:

Reviewer 2: **issue:** Avoid use of here to indicate position in the document. 
```suggestion
The following Flutter classes that allow you to
modify variable fonts:
```

Author: Yeah, both of these wordings are removed.

* [`FontFeature`][] allows you to select glyphs in a font
* [`FontVariation`][] allows you to modify a font's design axes

> ## More info

Reviewer 2: **nit:** This lacks agency and also abbreviates unncessarily.

Consider:
```suggestion
## To learn more
```

> If you would like to read about one engineer's experience
> customizing variable fonts and animating them as they
> morph, check out [Playful typography with Flutter][],
> a free article on Medium. The associated example also
> uses a custom shader.

Reviewer 2: **issue:** This reads with difficulty. Consider breaking it down into more sentences.
```suggestion
To read about one experience, check out [Playful typography with Flutter][].
This Medium article details how one engineer customized variable fonts
and animated them as they morph. The associated example also used a
custom shader.
```

This reduced readability from Grade 14 to Grade 7 with the same number of words.
